File: tools/extract_frontend_endpoints.py
import csv
import re
import sys
from pathlib import Path

root = Path(r"C:\DATA\Elite Clinic")
files = []
for base, pattern in [
    (root / "Frontend" / "actions", "*.ts"),
    (root / "Frontend" / "app", "*.tsx"),
    (root / "Frontend" / "lib", "*.ts"),
]:
    if base.exists():
        files.extend(sorted(base.rglob(pattern)))

string_re = re.compile(r"'(?:\\.|[^'\\])*'|\"(?:\\.|[^\"\\])*\"|`(?:\\.|[^`\\])*`", re.DOTALL)
rows = []

for file_path in files:
    try:
        text = file_path.read_text(encoding="utf-8", errors="ignore")
    except Exception:
        continue

    rel = file_path.relative_to(root).as_posix()
    for match in string_re.finditer(text):
        literal = match.group(0)[1:-1]
        if "/api/" in literal:
            line = text.count("\n", 0, match.start()) + 1
            rows.append((rel, line, literal))

out_path = root / "docs" / "spec-kit" / "ENDPOINT_MATRIX_FRONTEND_USED.csv"
out_path.parent.mkdir(parents=True, exist_ok=True)

with out_path.open("w", newline="", encoding="utf-8") as f:
    writer = csv.writer(f)
    writer.writerow(["File", "Line", "RouteLiteral"])
    writer.writerows(rows)

unique_count = len({r[2] for r in rows})

print(f"CSV_PATH:{out_path.as_posix()}")
print(f"TOTAL_MATCHES:{len(rows)}")
print(f"UNIQUE_ROUTE_LITERAL_COUNT:{unique_count}")
print("FIRST_30_ROWS:")
writer = csv.writer(sys.stdout)
writer.writerow(["File", "Line", "RouteLiteral"])
for row in rows[:30]:
    writer.writerow(row)

File: tools/extract_frontend_endpoints.js
const fs = require('fs');
const path = require('path');

const root = path.resolve('C:/DATA/Elite Clinic');
const targets = [
  ['Frontend/actions', '.ts'],
  ['Frontend/app', '.tsx'],
  ['Frontend/lib', '.ts']
];

function walk(dir, ext, acc = []) {
  if (!fs.existsSync(dir)) return acc;
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      walk(full, ext, acc);
    } else if (entry.isFile() && full.endsWith(ext)) {
      acc.push(full);
    }
  }
  return acc;
}

let files = [];
for (const [subPath, ext] of targets) {
  files = files.concat(walk(path.join(root, subPath), ext));
}
files.sort();

const stringRe = /'(?:\\.|[^'\\])*'|"(?:\\.|[^"\\])*"|`(?:\\.|[^`\\])*`/gs;
const rows = [];

for (const filePath of files) {
  let text;
  try {
    text = fs.readFileSync(filePath, 'utf8');
  } catch {
    continue;
  }

  stringRe.lastIndex = 0;
  let match;
  while ((match = stringRe.exec(text)) !== null) {
    const literal = match[0].slice(1, -1);
    if (literal.includes('/api/')) {
      const line = (text.slice(0, match.index).match(/\n/g) || []).length + 1;
      const rel = path.relative(root, filePath).split(path.sep).join('/');
      rows.push([rel, line, literal]);
    }
  }
}

const outPath = path.join(root, 'docs/spec-kit/ENDPOINT_MATRIX_FRONTEND_USED.csv');
fs.mkdirSync(path.dirname(outPath), { recursive: true });

function csvEscape(value) {
  const s = String(value);
  return /[",\n\r]/.test(s) ? '"' + s.replace(/"/g, '""') + '"' : s;
}

const allRows = [['File', 'Line', 'RouteLiteral'], ...rows];
fs.writeFileSync(outPath, allRows.map(r => r.map(csvEscape).join(',')).join('\n'), 'utf8');

const uniqueCount = new Set(rows.map(r => r[2])).size;

console.log(`CSV_PATH:${outPath.replace(/\\/g, '/')}`);
console.log(`TOTAL_MATCHES:${rows.length}`);
console.log(`UNIQUE_ROUTE_LITERAL_COUNT:${uniqueCount}`);
console.log('FIRST_30_ROWS:');
for (const row of allRows.slice(0, 31)) {
  console.log(row.map(csvEscape).join(','));
}
